cpp/src/hash/hashing.cu

```diff
@@ -38,6 +38,22 @@ bool md5_type_check(data_type dt)
   return !is_chrono(dt) && (is_fixed_width(dt) || (dt.id() == type_id::STRING));
 }
 
+template <typename IterType>
+std::vector<column_view> to_leaf_columns(IterType iter_begin, IterType iter_end)
+{
+  std::vector<column_view> leaf_columns;
+  for_each(iter_begin, iter_end, [&](column_view const& col) {
+    if (is_nested(col.type())) {
+      CUDF_EXPECTS(col.type().id() == type_id::STRUCT, "unsupported nested type");
+      auto child_columns = to_leaf_columns(col.child_begin(), col.child_end());
+      leaf_columns.insert(leaf_columns.end(), child_columns.begin(), child_columns.end());
+    } else {
+      leaf_columns.emplace_back(col);
```

Comment:
Should I be worried that the `else` uses emplacement while the `if` uses insertion?  I'm not sure how to understand this. Is `col` being moved while `child_columns` are being copied?

Reply:
`col` is being used as a copy-constructor argument and `child_columns` are being copied as they are inserted.  These are all dealing with `column_view` which owns no resources, so there shouldn't be any move semantics here.

Reply:
OK. I was hoping that somehow the two branches could be made more symmetric to avoid confusion.  But I looked at how to do it and I couldn't find a way.